### PR TITLE
Solution to review

### PR DESCRIPTION
**Pull request by bcherepakha**

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -7,6 +7,308 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Catalog</h1>
+    <header class="header">
+
+      <div class="header__logo"></div>
```
Comment (bcherepakha) [inline src/index.html:12]: The logo should be clickable and accessible (to screen readers, for example). 

A clickable logo that leads to the homepage is a standard and almost universal practice in web design for several reasons:

1. **Improved Navigation**: Users often expect clicking on the logo to take them back to the main page of the website. This enhances navigation and provides an easy way to return to the starting page.

2. **Logic and Expectations**: User expectations help create an intuitive user interface. The absence of a clickable logo can cause confusion.

3. **Convenience and Accessibility**: A clickable logo provides additional clickable space on the website, improving accessibility and convenience for users, especially on mobile devices.

4. **Meeting Search Engine Expectations**: Popular search engines like Google typically recommend making the logo clickable to facilitate indexing and understanding of the site's structure.

While there are no strict rules in web design, a clickable logo that returns to the homepage is a long-standing and well-recognized practice that enhances user experience and helps users quickly navigate to the site's starting page.

File: src/index.html [hunk 2]
```diff
@@ -7,6 +7,308 @@
...
+      </nav>
+    </header>
+
+    <main class="container">
+      <div class="card card--location" data-qa="card">
+        <img src="images/imac.jpeg"
+          alt="imac"
+          class="card__photo">
+
+        <h3 class="card__name">
```
Comment (bcherepakha) [inline src/index.html:54]: I see the `h3` header, but do not see the h1 and h2. Please use the correct structure of the heading

Using the correct header structure on a page is essential for several reasons:

1. **Accessibility**: Proper header structure ensures that content is accessible to all users, including those who rely on screen readers. Screen readers use headers to navigate and understand the content's hierarchy.

2. **SEO**: Search engines rely on header tags to understand the content and its hierarchy on a page. Using the correct structure can improve search engine optimization (SEO) and the visibility of your content in search results.

3. **Content Organization**: Headers provide a clear and logical organization of content. They help readers quickly grasp the main topics and subtopics of a page, making it easier to find the information they need.

4. **Readability**: Well-structured headers improve the overall readability of the page. They break up the content into digestible sections, making it more user-friendly.

5. **User Experience**: Users can skim a page more efficiently with the help of headers. They can quickly identify relevant sections and navigate to the parts of the page that interest them.

6. **Consistency**: Consistent use of headers throughout a website creates a uniform and predictable experience for users. They know what to expect in terms of content organization.

In summary, using the correct header structure not only enhances accessibility and SEO but also contributes to better content organization, readability, and overall user experience.

File: src/index.html [hunk 3]
```diff
@@ -7,6 +7,308 @@
...
+          Моноблок APPLE A1419 iMac 27" Retina 5K (MNED2UA/A)
+        </h3>
+        <p class="card__code">Код товара: 195434</p>
+
+        <div class="card__rating">
+          <div class="card__stars--4">
```
Comment (bcherepakha) [inline src/index.html:60]: Using a modifier without specifying the base class does not align with the principles of the BEM (Block Element Modifier) methodology

File: src/index.html [hunk 4]
```diff
@@ -7,6 +7,308 @@
...
+        <img src="images/imac.jpeg"
+          alt="imac"
+          class="card__photo">
+
+        <h3 class="card__name">
+          Моноблок APPLE A1419 iMac 27" Retina 5K (MNED2UA/A)
+        </h3>
+        <p class="card__code">Код товара: 195434</p>
+
+        <div class="card__rating">
```
Comment (bcherepakha) [inline src/index.html:59]: The layout of this rating does not meet accessibility requirements. Consider whether it can be fixed?

File: src/index.html [hunk 5]
```diff
@@ -7,6 +7,308 @@
...
+        </div>
+
+        <div class="card__price">
+          <p class="card__price__word">Цена:</p>
+          <p class="card__price__value">69 999 грн</p>
+        </div>
+
+        <a href="#"
+          class="card__button"
+          data-qa="card-hover">
```
Comment (bcherepakha) [inline src/index.html:110]: NB!: add `data-qa="card-hover"` to the link Buy inside **the first card**

File: src/index.html [hunk 6]
```diff
@@ -7,6 +7,308 @@
...
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+            <div class="stars__star"></div>
+          </div>
+          <p class="card__reviews">Отзывов: 5</p>
+        </div>
+
+        <div class="card__price">
+          <p class="card__price__word">Цена:</p>
+          <p class="card__price__value">69 999 грн</p>
```
Comment (bcherepakha) [inline src/index.html:105]: Why you think that this is paragraph?

File: src/index.html [hunk 7]
```diff
@@ -7,6 +7,308 @@
...
+        </div>
+
+        <a href="#"
+          class="card__button"
+          data-qa="card-hover">
+          Купить
+        </a>
+      </div>
+
+      <div class="card card--location" data-qa="card">
```
Comment (bcherepakha) [inline src/index.html:82]: NB! add `data-qa="card"` to **the first** card

File: src/index.html [hunk 8]
```diff
@@ -7,6 +7,308 @@
...
+            <a href="#" class="navigation__link">Видео</a>
+          </li>
+        </ul>
+      </nav>
+    </header>
+
+    <main class="container">
+      <div class="card card--location" data-qa="card">
+        <img src="images/imac.jpeg"
+          alt="imac"
```
Comment (bcherepakha) [inline src/index.html:51]: An image needs alternative text when it contains information or is an essential part of the content, and this information needs to be conveyed to users who cannot see the image, including those using screen readers, as well as for cases where the image cannot be loaded or displayed for some reason.

And what about that place?

File: src/styles/card.scss [hunk 9]
```diff
@@ -0,0 +1,110 @@
+.card {
+  width: 200px;
```
Comment (bcherepakha) [inline src/styles/card.scss:2]: In responsive design, it's not advisable to use absolute pixel-based sizes to specify the dimensions of a card. It's more effective to utilize flex layout properties. This approach allows for greater adaptability and responsiveness, ensuring that the card's size adjusts smoothly to different screen sizes and orientations. By avoiding fixed pixel sizes and embracing flexible layouts, your design can better accommodate various devices and screen dimensions, enhancing the overall user experience.

File: src/styles/card.scss [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+  padding: 15px 14px;
+
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+  font-family: "Roboto", sans-serif;
```
Comment (bcherepakha) [inline src/styles/card.scss:7]: It's preferable not to set the font family individually for each card; instead, it should inherit from the page styles. The exception is when a component requires a different, non-default style. This approach ensures a consistent and efficient use of typography across the page, with font settings inherited unless there's a specific need for a distinct style within the component.

File: src/styles/card.scss [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+
+  &__photo {
+    width: 160px;
+    height: 134px;
```
Comment (bcherepakha) [inline src/styles/card.scss:11]: It's more effective to specify a single size unit, such as width, and, if necessary, define height using the "aspect-ratio" property. This approach simplifies the layout and maintains the aspect ratio of the element, allowing for responsive design while ensuring consistent proportions.

File: src/styles/card.scss [hunk 12]
```diff
@@ -0,0 +1,110 @@
+.card {
+  width: 200px;
+  padding: 15px 14px;
+
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+  font-family: "Roboto", sans-serif;
+
+  &__photo {
+    width: 160px;
```
Comment (bcherepakha) [inline src/styles/card.scss:10]: Using absolute sizes here is not recommended. Consider whether relative units of measurement are more suitable. In web development, it's generally advisable to focus on describing the behavior of elements on the page rather than their exact dimensions. When referring to behavior, this entails specifying the relationship between the element's size and its container, employing flex or grid layout properties, and striving to align with guidelines or directions.

File: src/styles/card.scss [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+    height: 134px;
+    margin: 15px 5px 42px;
+  }
+
+  &__name {
+    margin: 0;
+    margin-bottom: 5px;
+
+    line-height: 15px;
+    font-size: 12px;
```
Comment (bcherepakha) [inline src/styles/card.scss:20]: Using pixels to define font properties is not recommended. Instead, use ratios, em, or rem units. This approach ensures that font sizes and properties are responsive and adapt to different screen sizes and text-scaling preferences, resulting in a more accessible and user-friendly typography.

File: src/styles/card.scss [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+
+  &__photo {
+    width: 160px;
+    height: 134px;
+    margin: 15px 5px 42px;
+  }
+
+  &__name {
+    margin: 0;
+    margin-bottom: 5px;
```
Comment (bcherepakha) [inline src/styles/card.scss:17]: These two entries about `margin` can be combined into one.

The use of "margin-bottom" is considered outdated. It's preferable to utilize block-level margin properties, such as "margin-block-end." This approach aligns with modern web design practices and enhances code clarity and compatibility.

File: src/styles/card.scss [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+    margin-bottom: 20px;
+  }
+
+  &__stars--4 {
+    display: flex;
+  }
+
+  .stars__star {
+    background: url(../images/star.svg) no-repeat center;
+    width: 15px;
```
Comment (bcherepakha) [inline src/styles/card.scss:43]: Using absolute sizes here is not recommended too.

File: src/styles/card.scss [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+  &__stars--1 .stars__star:nth-child(-n+1),
+  &__stars--2 .stars__star:nth-child(-n+2),
+  &__stars--3 .stars__star:nth-child(-n+3),
+  &__stars--4 .stars__star:nth-child(-n+4),
+  &__stars--5 .stars__star:nth-child(-n+5) {
+    background-image: url(../images/star-active.svg);
+  }
+
+  &__reviews {
+    display: inline-block;
```
Comment (bcherepakha) [inline src/styles/card.scss:61]: You do not need this display here. You do not need to change display here at all.

File: src/styles/card.scss [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+    &__value {
+      line-height: 19px;
+      font-size: 16px;
+      font-weight: 700;
+    }
+  }
+
+  &__button {
+    display: block;
+    width: 170px;
```
Comment (bcherepakha) [inline src/styles/card.scss:85]: Using absolute sizes here is not recommended too.

File: src/styles/card.scss [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+    font-size: 14px;
+    font-weight: 700;
+    color: #fff;
+    text-align: center;
+    text-transform: uppercase;
+    text-decoration: none;
+  }
+}
+
+.card--location {
```
Comment (bcherepakha) [inline src/styles/card.scss:103]: Why you do not use nesting here?

File: src/styles/header.scss [hunk 19]
```diff
@@ -0,0 +1,16 @@
+.header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  height: 60px;
```
Comment (bcherepakha) [inline src/styles/header.scss:5]: do not use fixed values for the height. Strive for the height to be determined by the content.

File: src/styles/header.scss [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.05);
+
+  &__logo {
+    position: relative;
+    width: 40px;
+    height: 40px;
+    left: 50px;
```
Comment (bcherepakha) [inline src/styles/header.scss:12]: do not use relative positioning for this reasons. Try to describe the behavior of elements and their relationships with each other.

File: src/styles/main.scss [hunk 21]
```diff
@@ -1,3 +1 @@
-body {
-  margin: 0;
-}
+@import "header", "navigation", "container", "card", "reset";
```
Comment (bcherepakha) [inline src/styles/main.scss:1]: Reflect on the impact that the order of imports can have and how accurate this sequence is.

File: src/styles/navigation.scss [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+
+  &__items-list {
+    display: flex;
+    align-items: center;
+    height: 60px;
+    margin: 0;
+    padding: 0;
+    list-style: none;
+
+    &__item {
```
Comment (bcherepakha) [inline src/styles/navigation.scss:13]: Here, when using nesting, it's evident that these class names don't conform to the BEM methodology.

File: src/index.html [hunk 23]
```diff
@@ -7,6 +7,308 @@
...
   </head>
   <body>
-    <h1>Catalog</h1>
+    <header class="header">
+
+      <div class="header__logo"></div>
+
+      <nav class="navigation">
+        <ul class="navigation__items-list">
+          <li class="navigation__items-list__item">
```
Comment (bcherepakha) [inline src/index.html:16]: these class names don't conform to the BEM methodology.
